### ISSUE TITLE
Make Tabulator non-editable by default.

### ISSUE THREAD

**Issue by MarcSkovMadsen** [holoviz/panel#4349]
For 95% of my use cases so far I don't need or want the user to edit the Tabulator table. I want them to view the table and sometimes select.

But Panel Tabulator is editable by default. This leads to several pains

- Bad UX: If a cell can be edited a user would expect this to mean something.
- I cannot easily figure out how to turn this off
- Instead what I would want in 50% of my use cases is that I can select a row by clicking any cell in a row. I cannot figure out how to turn this on.

Please 1) Make Tabulator non-editable by default 2) add a simple to use parameter to turn this on or off.

## Example

The below shows what for me are sensible defaults: `show_index=False` and enable selection of 1 row.

But I cannot select the row. And the cells are editable - which I do not expect.

https://user-images.githubusercontent.com/42288570/215165226-1d235b43-f1b4-4047-82d4-1806f1fc63b7.mp4

```python
import panel as pn
import pandas as pd
import datetime as dt

pn.extension("tabulator")

df = pd.DataFrame({
    'int': [1, 2, 3],
    'float': [3.14, 6.28, 9.42],
    'str': ['A', 'B', 'C'],
    'bool': [True, False, True],
    'date': [dt.date(2019, 1, 1), dt.date(2020, 1, 1), dt.date(2020, 1, 10)],
    'datetime': [dt.datetime(2019, 1, 1, 10), dt.datetime(2020, 1, 1, 12), dt.datetime(2020, 1, 10, 13)]
}, index=[1, 2, 3])

df_widget = pn.widgets.Tabulator(df, buttons={'Print': "<i class='fa fa-print'></i>"}, selectable=1, show_index=False)
pn.panel(df_widget).servable()
```

**Comment by MarcSkovMadsen** [comment]
## Workaround: Disable editing for all columns

Ok. Editing can be disable for each column by specifying `None  as the editor. I would say the intuitive thing is to have to enable editing for specific columns. Not that I have to spend time on disabling each column.

```python
import panel as pn
import pandas as pd
import datetime as dt

pn.extension("tabulator")

df = pd.DataFrame({
    'int': [1, 2, 3],
    'float': [3.14, 6.28, 9.42],
    'str': ['A', 'B', 'C'],
    'bool': [True, False, True],
    'date': [dt.date(2019, 1, 1), dt.date(2020, 1, 1), dt.date(2020, 1, 10)],
    'datetime': [dt.datetime(2019, 1, 1, 10), dt.datetime(2020, 1, 1, 12), dt.datetime(2020, 1, 10, 13)]
}, index=[1, 2, 3])

editors = {key: None for key in df}

df_widget = pn.widgets.Tabulator(df, buttons={'Print': "<i class='fa fa-print'></i>"}, selectable=1, show_index=False, editors=editors)
pn.panel(df_widget).servable()
```

**Comment by xavArtley** [comment]
As far as I remember enable=False is what you want

**Comment by philippjfr** [comment]
Correct, `disabled` is the canonical parameter to disable all Widget types.

**Comment by maximlt** [comment]
I'm in favor of closing this issue. There's a clear way to make the widget non-editable with `disabled`, and I believe widgets should all by default be editable. Having a pane based on Tabulator would be an alternative.

**Comment by MarcSkovMadsen** [comment]
A few comments

- What does `disabled` really mean? Is it the pagination, editing, ... or everything that should be disabled?
- As editing disabled is such a basic feature why is it not
  - the default?
  - documented?

**Comment by maximlt** [comment]
> What does disabled really mean? Is it the pagination, editing, ... or everything that should be disabled?

To me that means disabling editing.

> As editing disabled is such a basic feature why is it not
> 
>     * the default?
> 
>     * documented?

- I would find it very weird for the default value of a **widget** to be disabled. To me Tabulator=Table+Editor ;)
- It is documented here https://panel.holoviz.org/reference/widgets/Tabulator.html#display

Even if I disagree with making the tabulator widget non-editable by default, I think though that you raise a good point and that users want a nice table pane with pagination, etc.

**Comment by philippjfr** [comment]
I definitely think it makes sense for us to add a pane type for tabulator as an alternative to pn.pane.DataFrame. Main question is whether that should take precedence over the DataFrame html repr.

**Comment by jbednar** [comment]
A pane type for tabulator would be great. I'd also love to see something that formats tables nicely, like https://cran.r-project.org/web/packages/kableExtra/vignettes/awesome_table_in_html.html . Tabulator focuses on making larger tables easy to explore (with filtering, sorting, etc.), while it's also important just to be able to render a table nicely on the page, and with smaller tables none of those features are needed (and thus the widgets to control them can be omitted from the display).

**Comment by aeroaks** [comment]
Good to see someone has already reported this issue.

**Comment by contang0** [comment]
How can I make only some of the columns editable though? I tried this but it doesnt work:

```
tabulator_editors = {
    "col": {"disabled": False},
}

table = pn.widgets.Tabulator(
    df_table,
    editors=tabulator_editors,
    disabled=True,
)
```

**Comment by maximlt** [comment]
@contang0 this is [briefly mentioned in the docs](https://panel.holoviz.org/reference/widgets/Tabulator.html#editors-editing): `editors={'col': None}`.

**Comment by hmijail** [comment]
Found this issue after spending some time googling for a way to do this and nearly giving up for the `configuration` escape hatch.

While it's true that the docs do mention the `disabled` option, it's IMO impossible to find unless you know exactly what you're looking for.

To wit: if I want to disable **editing**, I have to **ignore** the "Editors/Editing" header, and instead go to the **"Display"** header, where the option is plainly named "disabled". 

Furthermore, it's misleading. As [asked](https://github.com/holoviz/panel/issues/4349#issuecomment-1453393070) by @MarcSkovMadsen , the next question I immediately got was "what exactly is being disabled, given that it's under the Display header?" Will it e.g. disable other interactivity like pagination? Will it grey out everything? But turns out that it doesn't even disable the row highlighting.
